internal/update/selfupdate.go

```
@@ -0,0 +1,175 @@
+		}
+		return nil
+	}
+	oldPath := exePath + ".old"
+	if err := deps.Rename(exePath, oldPath); err != nil {
+		return fmt.Errorf("update: rename %s -> %s: %w", exePath, oldPath, err)
+	}
+	if err := deps.Rename(newPath, exePath); err != nil {
+		// Roll the old binary back; if that also fails we can't fix it for
+		// the user so tell them where the last-known-good copy lives.
+		if rbErr := deps.Rename(oldPath, exePath); rbErr != nil {
+			return fmt.Errorf("update: replace %s failed (%w); rollback also failed (%v); recover from %s", exePath, err, rbErr, oldPath)
```

golangci-lint (2.11.4)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)